front/src/applications/operationalStudies/helpers/upsertNewProjectedTrains.ts

```diff
@@ -1,27 +1,42 @@
 import { type ProjectPathTrainResult } from 'common/api/osrdEditoastApi';
 import type { TrainSpaceTimeData } from 'modules/simulationResult/types';
-import type {
-  TimetableItemId,
-  TimetableItemWithTimetableId,
-  TrainScheduleId,
-} from 'reducers/osrdconf/types';
+import type { TimetableItemId, TimetableItemWithTimetableId } from 'reducers/osrdconf/types';
+import { Duration } from 'utils/duration';
+import {
+  formatPacedTrainIdToOccurrenceId,
+  isPacedTrainResponseWithPacedTrainId,
+} from 'utils/trainId';
 
 const upsertNewProjectedTrains = (
   projectedTrains: Map<TimetableItemId, TrainSpaceTimeData>,
-  projectedTrainsToUpsert: Map<TrainScheduleId, ProjectPathTrainResult>,
-  trainSchedulesById: Map<TimetableItemId, TimetableItemWithTimetableId>
+  projectedTrainsToUpsert: Map<TimetableItemId, ProjectPathTrainResult>,
+  timeTableItemsById: Map<TimetableItemId, TimetableItemWithTimetableId>
```

Comment:
We usually don't use a capital letter for `Table` in `timetable` (see in the code base)
```suggestion
  timetableItemsById: Map<TimetableItemId, TimetableItemWithTimetableId>
```